Beam/resources/assets/js/remplib.js

```diff
@@ -549,6 +575,64 @@ remplib = typeof(remplib) === 'undefined' ? {} : remplib;
             if (0 === this.totalTimeSpent % logInterval) {
                 this.trackTimespent();
             }
+        },
+
+        actualProgress: function() {
+            var h = document.documentElement,
+                b = document.body;
+
+            return (h.scrollTop||b.scrollTop) / ((h.scrollHeight||b.scrollHeight) - h.clientHeight);
+        },
+
+        scrollProgressEvent: function () {
+            var a = this.trackedArticle,
+                d = { document: this.actualProgress },
+                dt = document.documentElement.scrollTop;
+
+            if ( Math.abs( dt - this.dt ) < 50 ) {
```

Comment:
Please comment blocks of code to explain what they do. Looking at this I've no idea why it's there. Same for the block below starting at line 598.